src/fsd/4-entities/upgrade/upgrades.service.ts

```
@@ -23,6 +26,178 @@
     static readonly recipeDataFull: IRecipeDataFull = this.convertRecipeData();
+    static readonly recipeExpandedUpgradeData: IRecipeExpandedUpgradeData = this.expandRecipeData();
+    public static readonly materialByLabel: Record<string, string> = this.createMaterialByLabelLookup();
+
+    /**
+     * @returns a lookup table keyed by material ID or material label pointing
+     *          to the material ID.
+     */
+    private static createMaterialByLabelLookup(): Record<string, string> {
+        const result: Record<string, string> = {};
+
+        Object.entries(this.recipeExpandedUpgradeData).forEach(data => {
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)